impl/client/src/main/javascript/web/dojo/pentaho/common/overrides/dijit/Dialog.js

```diff
@@ -0,0 +1,674 @@
+			//
+			//		New dialog will be displayed on top of all currently displayed dialogs.
+			//
+			//		Caller is responsible for setting focus in new dialog after the fade-in
+			//		animation completes.
+
+			// Save current focus
+			ds[ds.length - 1].focus = focus.curNode;
+
+			// Set z-index a bit above previous dialog
+			var zIndex = ds[ds.length - 1].dialog ? ds[ds.length - 1].zIndex + 2 : Dialog._DialogLevelManager._beginZIndex;
+			domStyle.set(dialog.domNode, 'zIndex', zIndex);
+
+			// Display the underlay, or if already displayed then adjust for this new dialog
+			DialogUnderlay.show(underlayAttrs, zIndex - 1);
+
+			ds.push({dialog: dialog, underlayAttrs: underlayAttrs, zIndex: zIndex});
+		},
+
+		hide: function(/*dijit/_WidgetBase*/ dialog){
+			// summary:
+			//		Called when the specified dialog is hidden/destroyed, after the fade-out
+			//		animation ends, in order to reset page focus, fix the underlay, etc.
+			//		If the specified dialog isn't open then does nothing.
+			//
+			//		Caller is responsible for either setting display:none on the dialog domNode,
+			//		or calling dijit/popup.hide(), or removing it from the page DOM.
+
+			if(ds[ds.length - 1].dialog == dialog){
+				// Removing the top (or only) dialog in the stack, return focus
+				// to previous dialog
+
+				ds.pop();
+
+				var pd = ds[ds.length - 1];	// the new active dialog (or the base page itself)
+
+				// Adjust underlay
+				if(ds.length == 1){
+					// Returning to original page.  Hide the underlay.
+					DialogUnderlay.hide();
+				}else{
+					// Popping back to previous dialog, adjust underlay.
+					DialogUnderlay.show(pd.underlayAttrs, pd.zIndex - 1);
+				}
+
+				// Adjust focus.
+				// TODO: regardless of setting of dialog.refocus, if the exeucte() method set focus somewhere,
+				// don't shift focus back to button.  Note that execute() runs at the start of the fade-out but
+				// this code runs later, at the end of the fade-out.  Menu has code like this.
+				if(dialog.refocus){
+					var focus;
+					if(dialog.dialogOpener){
+						focus = dialog.dialogOpener;
+						dialog.dialogOpener = null;
+					}else{
+            // If we are returning control to a previous dialog but for some reason
+            // that dialog didn't have a focused field, set focus to first focusable item.
+            // This situation could happen if two dialogs appeared at nearly the same time,
+            // since a dialog doesn't set it's focus until the fade-in is finished.
+            focus = pd.focus;
+            if(pd.dialog && (!focus || !dom.isDescendant(focus, pd.dialog.domNode))){
+              pd.dialog._getFocusItems(pd.dialog.domNode);
+              focus = pd.dialog._firstFocusItem;
+					  }
+					}
+
+					if(focus){
+						// Refocus the button that spawned the Dialog.   This will fail in corner cases including
+						// page unload on IE, because the dijit/form/Button that launched the Dialog may get destroyed
```

Comment:
I believe formatting is completely off, compared to the original. Could you have auto-formatted the file? In my local version, I see a 2 space indented file. This change will make it harder (now and) in the future to know exactly what changed. And, in such a big file, it's important.

So, please, restore the original formatting and add your modifications following the original formatting.

Also, please, to the same end of making it easier to discover changes, add a comment before changed lines including the word "Pentaho": For example:
```js
// Pentaho - adding support for a fixed dialog opener.
```

Reply:
Also, some of the files within the overrides folder have a header such as:

```js
/* Overridden due to PRD-5440
		see lines: 200-205
*/
```

I'd use instead something like:
```js
/* Overridden due to PRD-5440
		see comments starting with // Pentaho
*/
```

Reply:
@AAravind27 I don't think that in your last commit you have restored the original formatting as I asked for:

> So, please, restore the original formatting and add your modifications following the original formatting.

Reply:
I checked the original file and it seems that, unlike our own codebase, it uses tabs instead of spaces, and that's why I thought it was looking strange.
Unfortunately, this also means that your new additions/changes would preferably use tabs to match.